instant-xml-macros/src/lib.rs

```diff
@@ -2,9 +2,39 @@ extern crate proc_macro;
 
 use proc_macro::TokenStream;
 use quote::quote;
+use std::collections::HashMap;
 use syn::{parse_macro_input, DeriveInput, Lit, Meta, NestedMeta};
 
-fn retrieve_default_namespace(input: &DeriveInput) -> Option<String> {
+type DefaultNamespace = String;
+type OtherNamespaces = HashMap<String, String>;
+type Namespaces = Option<(Option<DefaultNamespace>, Option<OtherNamespaces>)>;
+
+fn retrieve_namespaces(input: &DeriveInput) -> Namespaces {
+    let mut default_namespace = DefaultNamespace::default();
+    let mut other_namesapces = OtherNamespaces::default();
+
+    if let NestedMeta::Meta(Meta::List(list)) = get_meta_items(&input.attrs).first()? {
+        if list.path.get_ident()? == "namespace" {
+            for item in &list.nested {
+                match item {
+                    NestedMeta::Lit(Lit::Str(v)) => {
+                        default_namespace = v.value();
+                    }
+                    NestedMeta::Meta(Meta::NameValue(key)) => {
+                        if let Lit::Str(value) = &key.lit {
+                            other_namesapces
+                                .insert(key.path.get_ident().unwrap().to_string(), value.value());
+                        }
+                    }
+                    _ => (),
+                }
+            }
+        }
+    }
+    Some((Some(default_namespace), Some(other_namesapces)))
+}
+
+fn retrieve_field_namespace(input: &syn::Field) -> Option<String> {
```

Comment:
`retrieve_namespaces` and `retrieve_field_namespace` probability could be unified somehow, I will take a look at it after my leave.